src/pages/index.tsx

```diff
@@ -1,17 +1,77 @@
-import { Inter } from "next/font/google";
+import { AnimalPost } from "@/components/AnimalPost/AnimalPost";
+import { Sidebar } from "@/components/Sidebar/Sidebar";
+import { SidebarLink } from "@/components/Sidebar/SidebarLink";
+import { animals } from "@/mock";
+import { useEffect, useRef } from "react";
+import { throttle } from "throttle-debounce";
+import { v4 as uuid } from "uuid";
 
-const inter = Inter({ subsets: ["latin"] });
-import { LogoNerdbord } from "../components/LogoNerdbord";
+export default function Home() {
+  const cardRefs = useRef<Array<HTMLElement | null>>([]);
+  const linkRefs = useRef<Array<HTMLElement | null>>([]);
 
-import styles from "../styles/index.module.css";
+  const highlightCurrentAnimal = () => {
+    let current: string | null | undefined = "";
+
+    cardRefs.current.forEach((card) => {
+      const sectionTop = card?.offsetTop;
+      if (scrollY + 400 >= sectionTop!) {
+        current = card?.getAttribute("id");
+      }
+    });
+
+    linkRefs.current.forEach((link) => {
+      link?.children[0].classList.remove("active");
+      if (link?.attributes[0].value.includes(current!)) {
+        link?.children[0].classList.add("active");
+      }
+    });
+  };
+
+  const throttleHighlightCurrentAnimal = throttle(200, () => {
+    highlightCurrentAnimal();
+  });
+
+  useEffect(() => {
+    console.log(cardRefs.current);
+    console.log(linkRefs.current);
+    window.addEventListener("scroll", throttleHighlightCurrentAnimal);
+
+    return () => {
+      window.removeEventListener("scroll", throttleHighlightCurrentAnimal);
+    };
+  }, []);
 
-export default function Home() {
   return (
-    <div className={styles.wrapper}>
-      <LogoNerdbord />
-      <h1 className={inter.className}>
-        Recruitment task for Javascript Trainee
-      </h1>
+    <div className="background">
+      <div className="wrapper">
+        <Sidebar>
+          {animals.map((link, idx) => {
+            return (
+              <SidebarLink
+                key={uuid()}
+                ref={(ref) => (linkRefs.current[idx] = ref)}
+                header={link.header}
+                text={link.text}
+                img={link.img}
+              />
+            );
+          })}
+        </Sidebar>
+        <div>
+          {animals.map((card, idx) => {
+            return (
+              <AnimalPost
+                key={uuid()}
```

Comment:
cwaniaku to rozwiązanie jest gorsze niż index :D uuid wywołuje się przy każdym renderze więc zawsze masz inny key

Reply:
2 opcje:
- w mocku dodajesz atrybut id:uuid() ale wtedy i tak musisz sprawdzić czy co render nie masz nowych idików, jeśli tak to robisz np. React.memo (ale raczej idiki nie będą się zmieniać ;))
- const animalsWithId = React.memo(()=>animals.map((animal)=>({id:uuid(),...animal})),[])